services/QuillLMS/spec/services/classlink_integration/enrollment_persister_spec.rb

```diff
@@ -0,0 +1,156 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ClasslinkIntegration::EnrollmentPersister do
+  let(:incoming_student_enrollment) do
+    {
+      role: User::STUDENT,
+      user_sourced_id: 'student-1',
+      classroom_sourced_id: 'classroom-1',
+      status: 'active'
+    }
+  end
+
+  let(:incoming_teacher_enrollment) do
+    {
+      role: User::TEACHER,
+      user_sourced_id: 'teacher-1',
+      classroom_sourced_id: 'classroom-1',
+      status: 'active'
+    }
+  end
+
+  let(:classlink_instance) { create(:classlink_instance) }
+
+  describe '#run' do
+    context 'no user or classroom found' do
+      it 'should do nothing and return early' do
+        expect do
+          described_class.new(
```

Comment:
You can use `described.class.run` instead of `described_class.new(...).run` since this is an `ApplicationService`